doc/y_cable/design_doc.md

```diff
@@ -0,0 +1,226 @@
+## Multiple y cable vendor API design
+
+### Overview
+
+This document summarizes the approach taken to accommodate multiple
+Y cable vendors for supporting dual-ToR configurations in SONiC.
+All the design choices are presented
+such that vendors can implement the Y cable interface with ease and effeciency
+and all the feature requirements are met.
+
+
+Challenge: to provide an interface for Y cable to interact with PMON docker and HOST
+           which can be uniform across all vendors.
+
+### Directory Structure
+
+#### Background
+
+- We need to have an appropriate place where vendors can implement their modules/packages and provide Sonic with an abstraction layer where pmon and host can access the Y cable packages of all the vendors with ease
+   
+  - Currently the assumption is Y cable package implementation would be in python itself
+  - If a vendor provides shared library, that scenario would be considered accordingly
```

Comment:
Rephrase to state that if a vendor has an existing library in a different language, they will need to either find a way to wrap/bind it in Python to align with the description below or (preferrably) provide a pure Python implementation.

Reply:
fixed